### PR TITLE
docksal config and Github templates

### PR DESCRIPTION
**Pull request by jbickar**
# READY FOR REVIEW

# Summary
- Tweaks to Docksal config
- Add Github templates

# Needed By (Date)
- Before `SOE-SOEOP-368` gets merged into `soe-develop-mc`

# Criticality
- How critical is this PR on a 1-10 scale? 4/10
- Adds some helpful things for Docksal, and Github templates

# Steps to Test

1. Read through the changes
2. Test anything that you think looks incorrect
3. Merge

# Affected Projects or Products
- SoE school site

# Associated Issues and/or People
- JIRA ticket: https://stanfordits.atlassian.net/browse/SOEOP-368
- Other PRs: N/A
- Anyone who should be notified? ( @sherakama and @pookmish for visibility)

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

File: .docksal/README.md [hunk 1]
```diff
@@ -3,13 +3,14 @@
 ##Steps to setup
 
 1. clone repo -- will default SOE-develop-mc branch.
-2. Checkout branch, e.g., `git checkout -b SOE-jiraticket-shortdescription`
+2. Checkout new branch, e.g., `git checkout -b SOE-jiraticket-shortdescription`
 3. Edit the following files, which are located ./docksal
```
Comment (mdyoung3) [inline .docksal/README.md:7]: Maybe note that the developer can know the base_url by the name of the root folder. I.e., if the folder name is grunt-drupal-task, then base_url will be https://grunt-drupal-task.dockal.

**Review by mdyoung3 — COMMENTED** [review]
I'd kind of like to clarify the base url. See my note.

File: .docksal/README.md (hunk 1 above)
Comment (jbickar) [reply .docksal/README.md:7]: OK, I'll update the README and clean up merge conflicts.

**Comment by jbickar** [comment]
@mdyoung3 , should be ready for another review now.

**Comment by jbickar** [comment]
Didn't we talk about a way that `.docksal/docksal.env` can include a separate file that's not tracked in version control?

**Comment by jbickar** [comment]
Oh, maybe this: https://github.com/docksal/docksal/blob/develop/docs/content/stack/configuration-variables.md#docksal_environment

File: .docksal/README.md [hunk 2]
```diff
@@ -3,13 +3,14 @@
...
-    3. Add branch name to line six in Gruntfile.js.
-    4. Edit soe.aliases.drushrc.php, making sure the slugs path to build/html is correct.
-    5. Edit settings.local.php, updating the `base_url` var. 
-6. Run `fin init`
-    
+    1. Add branch name to line six in Gruntfile.js.
+    2. Edit soe.aliases.drushrc.php, making sure the slugs path to build/html is correct.
+    3. Optional: edit settings.local.php, updating the `base_url` var. Note that the `base_url` is set based on the directory name of this repository, e.g., `grunt-drupal-tasks.docksal`.
+4. If you use a key other than `id_rsa` to connect to Acquia, run `fin ssh-key add <keyname>` (e.g., `fin ssh-key add id_rsa_acquia_4096`)
```
Comment (jbickar) [inline .docksal/README.md:11]: This does not persist. I'm not sure yet what causes it to go away - stopping the Docksal application, destroying the containers, or restarting my machine.

**Comment by mdyoung3** [comment]
> Oh, maybe this: https://github.com/docksal/docksal/blob/develop/docs/content/stack/configuration-variables.md#docksal_environment

Yes. I was also going to post this: https://docs.docksal.io/stack/understanding-stack-config/#docksal-local

**Comment by mdyoung3** [comment]
@jbickar 

We could include this template: https://github.com/docksal/boilerplate-drupal8/blob/fc0134900f59ad6fc32321dbb314be0c29491250/.docksal/docksal.env

It comes with most of the configs commented out, but more importantly, it has the steps to create a default settings file.

File: .docksal/README.md (hunk 2 above)
Comment (mdyoung3) [reply .docksal/README.md:11]: @jbickar ,

You'll need to add these keys to the global environment file. 

> For example, assuming you have a private SSH key $HOME/.ssh/acquia_key, you would define a variable:
> SECRET_SSH_KEY_ACQUIA='acquia_key'

Source: https://blog.docksal.io/loading-ssh-keys-in-docksal-containers-c6a48944ccc8

Comment (jbickar) [reply .docksal/README.md:11]: Global environment file, which is tracked in Git, or in `.docksal/docksal-local.env`?

`.docksal/docksal-local.env` appears to be working; I will add that to `.gitignore`

Comment (mdyoung3) [reply .docksal/README.md:11]: @jbickar 

By the way, you can edit the global environment file from $HOME/.docksal/docksal.env

Comment (mdyoung3) [reply .docksal/README.md:11]: Sorry. I didn't see your comment before making my last one. See my last one for location of the global.

Also, you'll have to run this command: `fin project restart` after adding the keys.

**Review by mdyoung3 — APPROVED** [review]

**Pull request merged**